src/Renci.SshNet/PrivateKeyFile.cs

```diff
@@ -470,22 +467,54 @@ private ED25519Key ParseOpenSshV1Key(byte [] keyFileData, string passPhrase)
             int checkInt1 = (int)privateKeyReader.ReadUInt32();
             int checkInt2 = (int)privateKeyReader.ReadUInt32();
             if (checkInt1 != checkInt2)
+                throw new SshException("The checkints differed, the openssh key was not correctly decoded." + checkInt1 + " <->" + checkInt2);
+
+            //key type
+            var keyType = privateKeyReader.ReadString(Encoding.UTF8);
+
+            Key parsedKey;
+            byte[] publicKey;
+            byte[] unencryptedPrivateKey;
+            switch (keyType)
             {
-                throw new SshException("The checkints differed, the openssh key was not correctly decoded.");
-            }
+                case "ssh-ed25519":
+                    //public key
+                    publicKey = privateKeyReader.ReadBignum2().ToByteArray().Reverse();
 
-            //key type, we already know it is ssh-ed25519
-            privateKeyReader.ReadString(Encoding.UTF8);
+                    //private key
+                    unencryptedPrivateKey = privateKeyReader.ReadBignum2().ToByteArray().Reverse();
```

Comment:
Can we somehow avoid going through BigInteger?

Reply:
if not that, then is ToBigInteger2() in Common/Extensions.cs more appropriate?